src/components/commons/Menubar/Menubar.component.tsx

```diff
@@ -0,0 +1,33 @@
+import { Close, Dehaze } from '@mui/icons-material';
+import { default as MuiButton } from '@mui/material/Button';
+import { useState } from 'react';
+
+import { MenubarProps } from './Menubar.type';
+
+export const Menubar = ({
+  onClick,
+  variant = 'text',
+  disabled,
+  color = 'inherit',
+  ...rest
+}: MenubarProps) => {
+  const [isClicked, setIsClicked] = useState(false);
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setIsClicked(!isClicked);
+    onClick?.(event);
+  };
```

Comment:
もしこのレイヤーでonClickHandler定義するならこうかも
```suggestion
  const handleClick = useCallback((event: React.MouseEvent<HTMLButtonElement>) => {
    setIsClicked(!isClicked);
    onClick?.(event);
  }, [isClicked, onClick]):
```